### PR TITLE
[INLONG-4099][Manager] Support create Iceberg resource

### PR DESCRIPTION
**Pull request by woofyzhao**
Feature #4099

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/IcebergPartition.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.common.enums;
+
+import org.apache.inlong.manager.common.util.Preconditions;
+
+import java.util.Locale;
+
+public enum IcebergPartition {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/IcebergPartition.java:24]: Please add Java docs for classes, enums, and public / static methods, thanks.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/IcebergType.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    IcebergType(String type) {
+        this.type = type;
+    }
+
+    /**
+     * Get type from name
+     */
+    public static IcebergType forType(String type) {
+        for (IcebergType ibType : values()) {
+            if (ibType.getType().toLowerCase(Locale.ROOT).equals(type.toLowerCase(Locale.ROOT))) {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/IcebergType.java:56]: Maybe the `equalsIgnoreCase()` is a better way to check whether the two types were equals or not.

**Review by lucaspeng12138 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/pom.xml [hunk 3]
```diff
@@ -167,6 +167,55 @@
...
+            <groupId>org.apache.iceberg</groupId>
+            <artifactId>iceberg-hive-metastore</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iceberg</groupId>
+            <artifactId>iceberg-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hive</groupId>
+            <artifactId>hive-metastore</artifactId>
```
Comment (healchow) [inline inlong-manager/manager-service/pom.xml:185]: Please check the NOTICEs and LICENSEs of those dependencies.

Comment (healchow) [reply inlong-manager/manager-service/pom.xml:185]: I found those files in `licenses/inlong-manager`, there was no problem for me.

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**